internal/pkg/db/db.go

```diff
@@ -76,3 +80,28 @@ type Configuration struct {
 func MakeTimestamp() int64 {
 	return time.Now().UnixNano() / int64(time.Millisecond)
 }
+
+// GetDatabaseCredentials retrieves the credentials to be used to connect to the underlying database.
+// If security is enabled then the credentials will be retrieved from the secret store.
+func GetDatabaseCredentials(dic *di.Container, configuration config.DatabaseInfo) (config.Credentials, error) {
```

Comment:
Assumes "primary"; I wonder if it would be better to pass in a single database struct (as opposed to the entire map).

Reply:
I think we are limited in this regard since the `DatabaseInfo` is a type alias for a `map` where the map's key is a `string` and the value is an anonymous struct. This limits us since we cannot use that anonymous struct in the function signature. Do you see anything wrong with giving an identity to that anonymous struct, or see another way to achieve this?

https://github.com/edgexfoundry/edgex-go/blob/7d0b6b305c2451dd3125ab97d00e56bcb3189c1a/internal/pkg/config/types.go#L115-L122

Reply:
I can't conceive of any reason not to turn the anonymous struct into a named type.